interop/src/clients/InteropClient/InteropClient/InteropAction.swift

```diff
@@ -0,0 +1,259 @@
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+
+import Foundation
+
+enum InteropResponse: Codable {
+    case success(value: String)
+    case failure(message: String)
+}
+
+enum InteropAction {
+    case initMLS(clientId: Data, ciphersuite: UInt16)
+    case getKeyPackage(ciphersuite: UInt16)
+    case addClient(conversationId: Data, ciphersuite: UInt16, keyPackage: Data)
+    case removeClient(conversationId: Data, clientId: Data)
+    case processWelcome(welcomePath: URL)
+    case encryptMessage(conversationId: Data, message: Data)
+    case decryptMessage(conversationId: Data, message: Data)
+    case initProteus
+    case getPrekey(id: UInt16)
+    case sessionFromPrekey(sessionId: String, prekey: Data)
+    case sessionFromMessage(sessionId: String, message: Data)
+    case encryptProteusMessage(sessionId: String, message: Data)
+    case decryptProteusMessage(sessionId: String, message: Data)
+    case getFingerprint
+}
+
+extension InteropAction {
+    init?(url: URL) {
+        switch url.host() {
+        case "init-mls":
+            let components = URLComponents(url: url, resolvingAgainstBaseURL: false)
+            let clientId = components?.queryItems?.first(where: {
+                $0.name == "client"
+            })?.value.flatMap {
+                Data(base64Encoded: $0)
+            }
+            let ciphersuite = components?.queryItems?.first(where: {
+                $0.name == "ciphersuite"
+            })?.value.flatMap {
+                UInt16($0)
+            }
+            
+            if let clientId, let ciphersuite {
+                self = .initMLS(clientId: clientId, ciphersuite: ciphersuite)
+            } else {
+                return nil
+            }
+        case "get-key-package":
+            let components = URLComponents(url: url, resolvingAgainstBaseURL: false)
+            let ciphersuite = components?.queryItems?.first(where: {
+                $0.name == "ciphersuite"
+            })?.value.flatMap { UInt16($0) }
+            
+            if let ciphersuite  {
+                self = .getKeyPackage(ciphersuite: ciphersuite)
+            } else {
+                return nil
+            }
+        case "add-client":
```

Comment:
There seems to be inconsistent formatting here -- no blank line between case blocks, whereas in blocks below there is. Do we have a swift formatter/linter checking this on CI?

Reply:
No we don't

Reply:
But good idea, I'll add ticket in JIRA